include/seastar/http/internal/content_source.hh

```diff
@@ -40,10 +40,15 @@ namespace internal {
  * */
 class content_length_source_impl : public data_source_impl {
     input_stream<char>& _inp;
+    http::reply* _rep;
```

Comment:
Let's make it symmetrical to `http_content_length_data_sink_impl` which doesn't reference the request itself, but directly the bytes-written counter on it